Sming/Components/AxtlsSsl/src/SslExtensionImpl.h

```diff
@@ -0,0 +1,42 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * AxtlsExtension.h
+ *
+ * SSL Extension implementation based on axTLS library
+ *
+ * @author: 2019 - Slavey Karadzhov <slav@attachix.com>
+ *
+ ****/
+
+#pragma once
+
+#include <Network/Ssl/SslInterface.h>
+#include <axtls-8266/compat/lwipr_compat.h>
+
+class SslExtensionImpl : public SslExtension
+{
+public:
+	SslExtensionImpl();
+
+	bool setHostName(const String& hostName) override;
+
+	bool setMaxFragmentSize(const SslExtensionFragmentSize& fragmentSize) override;
+
+	void* getInternalObject() override
+	{
+		return sslExtension;
+	}
+
+	virtual ~SslExtensionImpl()
+	{
+		free(sslExtension);
```

Comment:
Call to `ssl_ctx_free` frees all contained connections including any SSL_EXTENSIONS structure pointers, so freeing it in the destructor here isn't safe. Can remove this class entirely if we simplify as outlined above.